### PR TITLE
detect-flowbits: add details for flowbits v8

### PR DESCRIPTION
**Pull request by hadiqaalamdar**
Task #6309

- [x] I have read the contributing guide lines at https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/6309) ticket: https://redmine.openinfosecfoundation.org/issues/6309
Previous PR: https://github.com/OISF/suricata/pull/10018

Describe changes:
- added the recommended changes from the last PR. 
- Changed the is_or flag to bool. The operator string is working correctly for or operator.
- rebased with Shivani's PR: https://github.com/OISF/suricata/pull/10021

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1532

### CODE REVIEW

**Review by jufajardini — CHANGES_REQUESTED** [review]
Looking good to me, but the intermediary commit that still contains the `NOALERT` cmd type isn't necessary - as leads to the CI check. Could you please squash the commits?
I'll leave to Shivani the part about how to handle the `and` cases, as she'll probably be able to give insight on that much faster.

File: src/detect-engine-analyzer.c [hunk 1]
```diff
@@ -861,6 +863,47 @@ static void DumpMatches(RuleAnalyzer *ctx, JsonBuilder *js, const SigMatchData *
                 jb_close(js);
                 break;
             }
+            case DETECT_FLOWBITS: {
+                const DetectFlowbitsData *cd = (const DetectFlowbitsData *)smd->ctx;
+
+                jb_open_object(js, "flowbits");
+                switch (cd->cmd) {
+                    /* noalert has been removed and never gets to DumpMatches */
```
Comment (inashivb) [inline src/detect-engine-analyzer.c:871]: we haven't like removed anything. It just wasn't supposed to be here so we can remove this comment, I think.

File: src/detect-engine-analyzer.c [hunk 2]
```diff
@@ -861,6 +863,47 @@ static void DumpMatches(RuleAnalyzer *ctx, JsonBuilder *js, const SigMatchData *
...
+                } else if (cd->or_list_size > 0) {
+                    is_or = true;
+                    for (uint8_t i = 0; i < cd->or_list_size; i++) {
+                        const char *varname =
+                                VarNameStoreSetupLookup(cd->or_list[i], VAR_TYPE_FLOW_BIT);
+                        jb_append_string(js, varname);
+                    }
+                }
+                jb_close(js); // array
+                if (is_or == true) {
```
Comment (inashivb) [inline src/detect-engine-analyzer.c:901]: nit: this could just be `if (is_or)`

**Review by inashivb — CHANGES_REQUESTED** [review]
Nice work, Hadiqa! :)
Some minor comments inline. looking good from first sight.
1. Please squash commits as asked by Juliana.
2. Leave `and` for now. It is ok. It's not a separate command like `or` anyway.
3. Handle nits.

Make sure to submit clean PRs for both s-v and suricata. It'll make easier for us to review. Thanks!

**Comment by hadiqaalamdar** [comment]
New PR: https://github.com/OISF/suricata/pull/10044

**Pull request closed without merging**